src/BudgetManager/BudgetManager.CQRS/Handlers/CategoryHandlers/UpdateCategoryHandler.cs

```diff
@@ -10,32 +10,41 @@ namespace BudgetManager.CQRS.Handlers.CategoryHandlers
 {
     public class UpdateCategoryHandler : IRequestHandler<UpdateCategoryCommand, CategoryResponse>
     {
-        private readonly IBaseRepository<Category> _categoryRepository;
+        private readonly IBaseRepository<User> _userRepository;
         private readonly IMapper _mapper;
 
-        public UpdateCategoryHandler(IBaseRepository<Category> categoryRepository, IMapper mapper)
+        public UpdateCategoryHandler(IBaseRepository<User> userRepository, IMapper mapper)
         {
-            _categoryRepository = categoryRepository;
+            _userRepository = userRepository;
             _mapper = mapper;
         }
 
         public async Task<CategoryResponse> Handle(UpdateCategoryCommand request, CancellationToken cancellationToken)
         {
             var updateCategoryObject = request.updateCategoryObject;
+            var userId = request.updateCategoryObject.UserId;
+            var user = await _userRepository.FindByIdAsync(userId, cancellationToken);
+
+            if (user is null) { return null; }
 
             var mappedCategory = _mapper.Map<Category>(updateCategoryObject);
+            var userCategory = user.Categories.Where(c => c.Id == mappedCategory.Id).FirstOrDefault();
+            var index = user.Categories.IndexOf(userCategory);
+
+            if (userCategory is not null && index is not -1)
+            {
+                mappedCategory.Id = userCategory.Id;
+                user.Categories[index] = mappedCategory;
+
+                var filter = Builders<User>.Filter.Eq(u => u.Id, userId);
+                var update = Builders<User>.Update.Set(u => u.Categories, user.Categories);
```

Comment:
if user has 50 or 60 categories, how long will it take to update. what about transactions and notification. after one year user may have 10000 if not 1 million data for transaction. app will lag in this case

Reply:
https://www.mongodb.com/community/forums/t/mongodb-update-operation-performance/11676

Reply:
> if user has 50 or 60 categories, how long will it take to update. what about transactions and notification. after one year user may have 10000 if not 1 million data for transaction. app may lag in this case

Any alternative solutions?